src/fn/upload-extension-libs.js

```diff
@@ -0,0 +1,55 @@
+const util = require('util');
+const fs = require('fs');
+
+const readdir = util.promisify(fs.readdir);
+const stat = util.promisify(fs.stat);
+
+const environment = require('../lib/environment');
+const pouch = require('../lib/db');
+const log = require('../lib/log');
+const attachmentsFromDir = require('../lib/attachments-from-dir');
+const warnUploadOverwrite = require('../lib/warn-upload-overwrite');
+const insertOrReplace = require('../lib/insert-or-replace');
+
+const DIR_NAME = 'extension-libs';
+const DOC_ID = 'extension-libs';
+
+const getConfiguredLibs = async (dir) => {
+  try {
+    const stats = await stat(dir, fs.constants.R_OK);
+    if (!stats.isDirectory()) {
+      return []; // file, not directory
+    }
+    return await readdir(dir);
+  } catch(e) {
+    return []; // no readable configuration directory
```

Comment:
My opinion is the `sync` fs functions are an anti-pattern, because they are async but it's not clear to the caller. As above, once we drop legacy Node I'm hoping we can get rid of all sync calls, but open to debate on this!

Reply:
:+1: I am fine with ditching the `Sync` calls (though I did not realize until now how janky they are...)! Here I was more interested in not using the try-catch as part of the intended control flow....  This code has now been refactored anyway, so nothing else needs to happen here.  

Also, for anyone else who might stumble on this comment (aka future me) and is wondering about: `sync fs functions are an anti-pattern, because they are async but it's not clear to the caller.` apparently, these functions just straight up [block the thread until they are done](https://advancedweb.hu/do-not-use-fs-sync-methods-in-javascript-use-fs-promises-instead/).  So, they will not give you race-conditions or anything, but they are bad for performance and absolutely an anti-pattern.

Reply:
I've sort of ignored them in this case (and in scripts) because it's a CLI and the thread isn't going to be doing anything else, but definitely need to avoid uses in other services.